File: common/src/main/java/farm/rosehearth/compatemon/modules/compatemon/IPokemonEntityExtensions.java
package farm.rosehearth.compatemon.modules.compatemon;

import com.cobblemon.mod.common.api.reactive.ObservableSubscription;
import com.cobblemon.mod.common.entity.EntityProperty;
import net.minecraft.nbt.CompoundTag;
import net.minecraft.network.syncher.EntityDataAccessor;

public interface IPokemonEntityExtensions {
	
	abstract EntityDataAccessor<CompoundTag> getPERSISTENT_DATA();
	abstract void setPERSISTENT_DATA(EntityDataAccessor<CompoundTag> value);
	
	abstract EntityProperty<CompoundTag> compatemon$getPersistentData();
	abstract void compatemon$setPersistentData(EntityProperty<CompoundTag> value);
	
	abstract ObservableSubscription<?> getSubscriptions();
}

File: common/src/main/kotlin/farm/rosehearth/compatemon/modules/compatemon/IPokemonEntityExtensions.kt
package farm.rosehearth.compatemon.modules.compatemon

import com.cobblemon.mod.common.entity.EntityProperty
import com.cobblemon.mod.common.entity.pokemon.PokemonEntity
import net.minecraft.nbt.CompoundTag
import net.minecraft.network.syncher.EntityDataSerializers
import net.minecraft.network.syncher.SynchedEntityData

interface IPokemonEntityExtensions {

	abstract fun getPersistentData(): EntityProperty<CompoundTag>;
	abstract fun setPersistentData(p:EntityProperty<CompoundTag>);

	companion object{
		@JvmStatic val PERSISTENT_DATA = SynchedEntityData.defineId(PokemonEntity::class.java, EntityDataSerializers.COMPOUND_TAG)
	}
}